### PR TITLE
JBTM-2881 Do not hide application exceptions

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.jboss.org/browse/JBTM-2881

!PERF !QA_JTA !QA_JTS_JDKORB !BLACKTIE !QA_JTS_JACORB !QA_JTS_OPENJDKORB !XTS !RTS 
 NO_WIN !RTS

### CODE REVIEW

File: STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java [hunk 1]
```diff
@@ -400,8 +401,12 @@ public Object invoke (Object proxy, java.lang.reflect.Method method, Object[] ar
                             }
                         }
                     }
-    
-                    return method.invoke(_theObject, args);
+
+                    try {
+                        return method.invoke(_theObject, args);
+                    } catch (InvocationTargetException e) {
+                        throw e.getCause() != null ? e.getCause() : e;
```
Comment (ochaloup) [inline STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java]: would not be good to log the exception with some contextual information (what was invoked with what parameters) before the exception is thrown?

Comment (mmusgrov) [inline STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java]: No, this is application code throwing an exception. It is up to the developer to catch his own exception and log accordingly.

Comment (ochaloup) [inline STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java]: I meant log,  e.g. to debug, for easier further investigation. But it was just a question to think about.

Comment (mmusgrov) [inline STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java]: OK good point, done.

**Pull request merged**